lldb/source/Plugins/Language/CPlusPlus/GenericStringView.cpp

```diff
@@ -0,0 +1,131 @@
+//===-- GenericStringView.cpp --------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===---------------------------------------------------------------------===//
+
+#include "Generic.h"
+#include "LibCxx.h"
+
+#include "lldb/ValueObject/ValueObject.h"
+#include "lldb/lldb-forward.h"
+
+#include "Plugins/Language/CPlusPlus/CxxStringTypes.h"
+#include <tuple>
+
+using namespace lldb;
+using namespace lldb_private;
+using namespace lldb_private::formatters;
+
+using StringElementType = StringPrinter::StringElementType;
+
+enum class StlType {
+  LibCxx,
+  LibStdcpp,
+  MsvcStl,
+};
+
+std::tuple<StlType, lldb::ValueObjectSP, lldb::ValueObjectSP>
+extractStringViewData(ValueObject &valobj) {
```

Comment:
Could you extract the return into a `struct` or mention what each `ValueObjectSP` is?

This function should probably be `static`.